lib/shelley/bench/Restore.hs

```diff
@@ -223,86 +225,70 @@ cardanoRestoreBench tr c socketFile = do
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletSeq "0-percent-seq" $ mkSeqAnyState' @0 networkProxy)
-            benchmarksSeq
-
-        , bench_restoration @_ @ByronKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletRnd "0-percent-rnd" $ mkRndAnyState @0)
-            benchmarksRnd
 
-        , bench_restoration @_ @ByronKey
+    let benchRestoreMultipleWallets nWallets = bench_restoration @_ @ShelleyKey
             networkProxy
             (trMessageText tr)
             socketFile
             np
             vData
-            (walletRnd "0.1-percent-rnd" $ mkRndAnyState @1)
-            benchmarksRnd
-
-        , bench_restoration @_ @ByronKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletRnd "0.2-percent-rnd" $ mkRndAnyState @2)
-            benchmarksRnd
+            (map (\i -> walletSeq
+                ("w" <> T.pack (show i)) $ mkSeqAnyState' (Proxy @0) networkProxy)
+                [0..nWallets :: Int])
+            True -- Don't write progress to .timelog file(s)
+            (unsafeMkPercentage 0.01)
+            benchmarksSeq
 
-        , bench_restoration @_ @ByronKey
+    let benchRestoreRndWithOwnership permillProxy = bench_restoration
             networkProxy
             (trMessageText tr)
             socketFile
             np
             vData
-            (walletRnd "0.4-percent-rnd" $ mkRndAnyState @4)
+            [walletRnd (showPercent permillProxy <> "-percent-rnd")
+                $ mkRndAnyState' permillProxy networkProxy]
+            True -- Write progress to .timelog file
+            (unsafeMkPercentage 1)
             benchmarksRnd
 
-         , bench_restoration @_ @ShelleyKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletSeq "0.1-percent-seq" $ mkSeqAnyState' @1 networkProxy)
-            benchmarksSeq
-
-         , bench_restoration @_ @ShelleyKey
+    let benchRestoreSeqWithOwnership permillProxy = bench_restoration
             networkProxy
             (trMessageText tr)
             socketFile
             np
             vData
-            (walletSeq "0.2-percent-seq" $ mkSeqAnyState' @2 networkProxy)
+            [walletSeq (showPercent permillProxy <> "-percent-seq")
+                $ mkSeqAnyState' permillProxy networkProxy]
+            True -- Write progress to .timelog file
+            (unsafeMkPercentage 1)
             benchmarksSeq
 
-         , bench_restoration @_ @ShelleyKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletSeq "0.4-percent-seq" $ mkSeqAnyState' @4 networkProxy)
-            benchmarksSeq
+    runBenchmarks
+        [ benchRestoreMultipleWallets 1
```

Comment:
Takes 6s to reach 1%.

I suspect the extrapolated value (600s to 100%) is a bit wrong.

- Could increase the number of wallets for this test only
- Could leave it as is
- Could increase the restore-to target for this test only
- Could increase restore-to target for all {1,10,100} cases (but 100 would be really slow)

Reply:
Locally 1 wallet seq 0% takes 276.4 s, vs 600s extrapolated, so pretty wrong, yes.

Reply:
so what's executive summary of the above?

Reply:
Summary:
1. Should restore the 1 wallet to 10% for better extrapolation. But we can keep 10 and 100 wallets restoring to 1%, where it doesn't matter much.
2. There may be some interesting slowdowns (e.g. 10%, 16%), only visible when restoring to 100% (as Piotr also suggested), but I think we can ignore that in this PR.

Based on the data:
- To 1%
    - 1 wallets -> 4.015s (extrapolation is 54% higher than the to-10% value ❌)
    - 10 wallets -> 24.62s (extrapolation is 0.9% higher than the to-10% value 🆗 )
    - 100 wallets -> 345.0s (extrapolation is 0.8% higher than the to-10% value 🆗 )
- To 10%
    - 1 wallets -> 26.13s (extrapolation is 9.3% lower than the to-100% value 🤔 )
    - 10 wallets -> 244.0s (extrapolation is 14.1% lower than the to-100% value 🤔 )
    - 100 wallets -> 3423s (IIRC)
- To 100% (includes the Shelley part of the chain)
    - 1 wallets -> 288s
    - 10 wallets -> 2842s
    - 100 wallets -> _would take a long time to run_

Reply:
Thanks!